inc/ConfigMap.h

```diff
@@ -24,12 +25,27 @@
 #include <vector>
 
 namespace PCOE {
-    class ConfigMap : public std::unordered_map<std::string, std::vector<std::string>> {
+    class ConfigMap {
     public:
         ConfigMap() = default;
         ConfigMap(const std::string& filename);
         ConfigMap(const int argc, char* argv[]);
 
+        /**
+         * @brief       Gets string vector at given key
+         * @param       key     The key from which the vector is retrieved
+         * @return      The string vector corresponding to the given key
+         */
+        const std::vector<std::string>& getVector(const std::string key) const;
+
+        std::vector<std::string>& getVector(const std::string key);
+
+        double getDouble(const std::string key) const;
+
+        std::uint64_t getU64(const std::string key) const;
+
+        int getI32(const std::string key) const;
```

Comment:
I think this might be confusing to some users. Could we either call this getInt or getInt32? Also, I think we would want to get long, so call it getLong or getInt64? 

If you go with getInt64 make sure it returns std::int64_t